runtime/caml/platform.h

```diff
@@ -481,8 +481,8 @@ Caml_inline void caml_plat_lock_non_blocking(caml_plat_mutex* m)
 
 Caml_inline void caml_plat_unlock(caml_plat_mutex* m)
 {
-  DEBUG_UNLOCK(m);
   check_err("unlock", pthread_mutex_unlock(m));
+  DEBUG_UNLOCK(m);
```

Comment:
What is the rationale for this change? It looks wrong to me.

Reply:
`DEBUG_UNLOCK` decrements a thread local counter that holds the current number of locks held. If it is called before `pthread_mutex_unlock` and the latter fails, the count will have been lowered and there's a fatal error (thanks to `check_err`
If we want to inspect the state of the runtime with this debug counter, I think it makes more sense to decrement it iff `pthread_mutex_unlock` succeeded. 
Or is there something I missed?

Reply:
But if the unlock succeeds, you're opening a race where another thread can suddenly fail `caml_plat_assert_all_locks_unlocked()` between the unlock and the `DEBUG_UNLOCK` adjustment.

Reply:
You're right. I'll fix my Windows code too.
Unfortunately I don't have access to a computer until next Monday, so I can't remove this patch from the PR.

Reply:
Noprob - just make sure noone puts a "merge me" label on the PR. Or mark it as draft, which will prevent it from being merged by mistake.

Reply:
But isn't there also a race in the current code? If thread A calls `DEBUG_UNLOCK` and thread B calls the all unlocked assertion before `pthread_unlock` then there's also an inconsistency.

Reply:
Yes. Welcome to the joys of parallel programming. You can't have a reliable check without support from the OS (or at least, from the threads library). But narrower races are preferrable here.

Reply:
I've removed this change.